### PR TITLE
COMMERCE-8443 Add product name and description search to shop by diagram

### PR DESCRIPTION
**Pull request by dsitu**
https://issues.liferay.com/browse/COMMERCE-8443

### CODE REVIEW

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/search/spi/model/index/contributor/CSDiagramEntryModelDocumentContributor.java [hunk 1]
```diff
@@ -39,6 +46,58 @@ public void contribute(Document document, CSDiagramEntry csDiagramEntry) {
...
+			String description = cpDefinition.getDescription(languageId);
+			String name = cpDefinition.getName(languageId);
+			String shortDescription = cpDefinition.getShortDescription(
+				languageId);
+
+			if (languageId.equals(cpDefinitionDefaultLanguageId)) {
+				document.addText(Field.DESCRIPTION, description);
+				document.addText(Field.NAME, name);
+				document.addText(CPField.SHORT_DESCRIPTION, shortDescription);
+			}
```
Comment (ivicac) [inline modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/search/spi/model/index/contributor/CSDiagramEntryModelDocumentContributor.java:75]: hey @dsitu this block and the block below seems to me the same?

File: modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/search/spi/model/index/contributor/CSDiagramEntryModelDocumentContributor.java [hunk 2]
```diff
@@ -39,6 +46,58 @@ public void contribute(Document document, CSDiagramEntry csDiagramEntry) {
...
+		}
+
+		document.addText(
+			Field.NAME, cpDefinition.getName(cpDefinitionDefaultLanguageId));
+		document.addText(
+			Field.DESCRIPTION,
+			cpDefinition.getDescription(cpDefinitionDefaultLanguageId));
+		document.addText(
+			CPField.SHORT_DESCRIPTION,
+			cpDefinition.getShortDescription(cpDefinitionDefaultLanguageId));
```
Comment (ivicac) [inline modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/search/spi/model/index/contributor/CSDiagramEntryModelDocumentContributor.java:97]: hey @dsitu this block and the block above seems to me the same? Also NAMe, should come after DESCRIPTION.

File: modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/search/spi/model/index/contributor/CSDiagramEntryModelDocumentContributor.java (hunk 1 above)
Comment (dsitu) [reply modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/search/spi/model/index/contributor/CSDiagramEntryModelDocumentContributor.java:75]: oh yeah, taking a closer look, they are adding the same thing. I copied this from cpDefinitionIndexer, ill change them both to keep it consistent. Thanks!

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:sf

**Comment by ivicac** [comment]
ci:test:relevant

**Comment by ivicac** [comment]
ci:test:sf

**Comment by ivicac** [comment]
@dsitu can you please fix SF?

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/search/spi/model/query/contributor/CSDiagramEntryKeywordQueryContributor.java [hunk 3]
```diff
@@ -51,10 +51,18 @@ public void contribute(
...
 		_queryHelper.addSearchTerm(
 			booleanQuery, searchContext, Field.ENTRY_CLASS_PK, false);
+		_queryHelper.addSearchLocalizedTerm(
+			booleanQuery, searchContext, Field.NAME, false);
+		_queryHelper.addSearchTerm(
+			booleanQuery, searchContext, Field.NAME, false);
 		_queryHelper.addSearchTerm(
 			booleanQuery, searchContext, "sequence", false);
+		_queryHelper.addSearchLocalizedTerm(
+			booleanQuery, searchContext, CPField.SHORT_DESCRIPTION, false);
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/search/spi/model/query/contributor/CSDiagramEntryKeywordQueryContributor.java:65]: @dsitu @ivicac maybe I'm wrong, but I remember that search should be run also in short description, so I believe also SHORT_DESCRIPTION should be added with addSearchTerm to query helper..

**Review by andrea-ale-sbarra — COMMENTED** [review]
@ivicac can we sync on the short description before forwarding this?

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/search/spi/model/query/contributor/CSDiagramEntryKeywordQueryContributor.java (hunk 3 above)
Comment (dsitu) [reply modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/search/spi/model/query/contributor/CSDiagramEntryKeywordQueryContributor.java:65]: hmm, good point. I went ahead and added short description to regular search term. Thanks!

**Comment by andrea-ale-sbarra** [comment]
ci:forward

**Pull request closed without merging**